projects/js-packages/charts/src/components/private/radial-wipe-animation/radial-wipe-animation.tsx

```diff
@@ -0,0 +1,89 @@
+ * - Initially, the entire circle is drawn with a white mask.
+ * - The circle's border (stroke) is drawn with a black stroke, wide enough to cover the circle.
+ * - The stroke is then "un-drawn," creating the wipe animation.
+ * - A white mask makes the area visible, while a black mask makes the area invisible.
+ */
+
+/**
+ * RadialWipeProps - SVG mask props that requires a radial wipe animation effect.
+ *
+ * @param {object}                            props                         - The properties object.
+ * @param {string}                            props.id                      - The unique ID for the mask.
+ * @param {number}                            props.radius                  - The outer radius of the radial wipe.
+ * @param {number}                            [props.innerRadius=0]         - The inner radius of the radial wipe.
+ * @param {number}                            [props.durationMs=1000]       - The duration of the animation in milliseconds.
+ * @param {number}                            [props.wipePercentage=100]    - The percentage of the wipe animation to complete.
+ * @param {'clockwise' | 'counter-clockwise'} [props.direction='clockwise'] - The direction of the wipe animation.
+ * @param {Angle}                             [props.startAngle='-90deg']   - The starting angle of the wipe animation.
+ *
+ * @return {JSX.Element} The radial wipe mask element.
+ */
+export type RadialWipeProps = {
+	id: string;
+	radius: number;
+	innerRadius?: number;
+	durationMs?: number;
+	startAngle?: Angle;
+	direction?: 'clockwise' | 'counter-clockwise';
+	wipePercentage?: number;
+};
+
+/**
+ * Renders a SVG mask that creates a radial wipe animation effect.
+ *
+ * @param {RadialWipeProps} props - Component props
+ * @return {JSX.Element} The rendered mask component
+ */
+function RadialWipe( {
+	id,
+	radius,
+	innerRadius = 0,
+	durationMs = 1000,
+	wipePercentage = 100,
+	direction = 'clockwise',
+	startAngle = '-90deg',
+}: RadialWipeProps ) {
+	const stroke = 'black'; // The stroke will be un-drawn, hence 'black' mask.
+
+	const strokeWidth =
+		( radius - innerRadius ) * 2 + // The stroke is centered on the circumference, so we need to double the width.
+		1; // Added 1 to prevent sub-pixel rendering issues.
+
+	const scaleY = direction === 'clockwise' ? -1 : 1;
+
+	const isValidWipePercentage = 0 < wipePercentage && wipePercentage <= 100;
+	const animationDuration = `${
+		// If wipePercentage is invalid, set animation duration to 0 to disable animation.
+		isValidWipePercentage ? durationMs * ( 100 / wipePercentage ) : 0
+	}ms`;
+
+	return (
+		<mask id={ id }>
+			<circle
+				cx={ 0 }
+				cy={ 0 }
+				r={ radius }
+				pathLength="100"
+				fill="white"
+				style={ {
+					strokeWidth,
+					stroke,
+					animationDuration,
+					transform: `rotate(${ startAngle }) scaleY(${ scaleY })`,
+				} }
+				className={ styles.radialWipeCircle }
+			/>
+		</mask>
+	);
+}
+
+export default RadialWipe;
```

Comment:
```suggestion
export default RadialWipeAnimation;
```

Related to comment about exports.